File: minto/2주차_수학/1929.js
let input = require('fs')
  .readFileSync('/dev/stdin')
  .toString()
  .split(' ')
  .map((item) => +item);

function countDecimal(a, b) {
  let arr = Array(b + 1)
    .fill(true)
    .fill(false, 0, 2);

  for (let i = 2; i * i <= b; i++) {
    if (arr[i]) {
      for (let j = i * i; j <= b; j += i) {
        arr[j] = false;
      }
    }
  }

  return arr.forEach((value, index) => {
    if (index >= a && value) {
      console.log(index);
    }
  });
}
countDecimal(input[0], input[1]);

/*
입력 값이 많으니 전체부터 소수를 찾으면 시간초과가 발생할 수 있으니
애라토스테네스의 체 방법을 이용해서 소수를 구했다. 
*/

File: ujin/2주차_수학/1929.py
N, M = map(int, input().split())

def is_prime(n):
    if n == 1:
        return False
    for j in range(2, int(n**0.5)+1):
        if n % j == 0:
            return False
    return True

for i in range(N, M+1):
    if is_prime(i):
        print(i)

'''
범위 내의 소수를 출력하는 문제이다.
모든 수를 검사하면 시간 초과가 날 수 있기 때문에 제곱근까지만 검사하도록 했다.
'''